src/pkg/bundler/common.go

```diff
@@ -0,0 +1,184 @@
+		bundler *Bundler
+	)
+	if bundler, err = New(cfg); err != nil {
+		message.Fatalf(err, "bundler unable to setup, bad config: %s", err.Error())
+	}
+	return bundler
+}
+
+// ClearPaths clears out the paths used by Bundler
+func (b *Bundler) ClearPaths() {
+	b.FS.ClearPaths()
+}
+
+// ValidateBundle validates the bundle
+func (b *Bundler) ValidateBundle() error {
+	if b.bundle.Metadata.Version == "" {
+		return fmt.Errorf("zarf-bundle.yaml is missing required field: metadata.version")
+	}
+	if b.bundle.Metadata.Name == "" {
+		return fmt.Errorf("zarf-bundle.yaml is missing required field: metadata.name")
+	}
+	if len(b.bundle.Packages) == 0 {
+		return fmt.Errorf("zarf-bundle.yaml is missing required list: packages")
+	}
+	for idx, pkg := range b.bundle.Packages {
+		if pkg.Repository == "" {
+			return fmt.Errorf("zarf-bundle.yaml .packages[%d] is missing required field: repository", idx)
+		}
+		if pkg.Ref == "" {
+			return fmt.Errorf("zarf-bundle.yaml .packages[%s] is missing required field: ref", pkg.Repository)
+		}
+		url := fmt.Sprintf("%s:%s", pkg.Repository, pkg.Ref)
+		// validate access to packages as well as components referenced in the package
+		remote, err := oci.NewOrasRemote(url)
+		if err != nil {
+			// remote performs access verification upon instantiation
+			return err
+		}
+		err = remote.PullPackageMetadata(b.FS.tmp.Base)
+		if err != nil {
+			return err
+		}
+		defer b.FS.ClearPaths()
+		// TODO: validate signatures here
+		// TODO: are we gonna re-sign the packages within a bundle?
+		requestedComponents := pkg.Components
+		if len(requestedComponents) > 0 {
+			zarfYAML := types.ZarfPackage{}
+			err := utils.ReadYaml(b.FS.tmp.ZarfYaml, &zarfYAML)
+			if err != nil {
+				return err
+			}
+			for _, component := range requestedComponents {
+				c := utils.Find(zarfYAML.Components, func(c types.ZarfComponent) bool {
+					return c.Name == component
+				})
+				if c.Name == "" {
+					return fmt.Errorf("zarf-bundle.yaml .packages[%s].components[%s] does not exist in upstream: %s", pkg.Repository, component, url)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// CalculateBuildInfo calculates the build info for the bundle
+//
+// this is mainly mirrored from packager.writeYaml()
+func (b *Bundler) CalculateBuildInfo() error {
+	now := time.Now()
+
+	// Just use $USER env variable to avoid CGO issue.
+	// https://groups.google.com/g/golang-dev/c/ZFDDX3ZiJ84.
+	// Record the name of the user creating the package.
+	if runtime.GOOS == "windows" {
+		b.bundle.Build.User = os.Getenv("USERNAME")
+	} else {
+		b.bundle.Build.User = os.Getenv("USER")
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+	b.bundle.Build.Terminal = hostname
+
+	// TODO: investigate the best way forward for determining arch
+	b.bundle.Metadata.Architecture = runtime.GOARCH
+	b.bundle.Build.Architecture = runtime.GOARCH
+
+	b.bundle.Build.Timestamp = now.Format(time.RFC1123Z)
+
+	b.bundle.Build.Version = config.CLIVersion
```

Comment:
since this reuses a lot of the logic from packager common it might be good to have the common bits in a separate struct / utility function.